src/helpers.R

```diff
@@ -6,4 +6,7 @@ file.writeText <- function(path, text) {
     f <- file(path)
     on.exit(close(f))
     writeLines(text, f)
-}
\ No newline at end of file
+}
+
+Counter <- methods::setRefClass("Counter",fields=list(iteration="numeric",
```

Comment:
I would avoid reference classes here.  But below is an R6 version of this that might be useful

```r
Counter <- R6::R6Class(
  "Counter",
  public = list(
    iteration = 0L,
    par = NULL,

    update = function(p) {
      if (isGradientStep(p, self$par)) {
        FALSE
      } else {
        self$par <- p
        TRUE
      }
    }
  ))    
```